src/openqaoa-core/qaoa_components/ansatz_constructor/gatemap.py

```diff
@@ -292,103 +369,100 @@ def _decomposition_standard(self) -> List[Tuple]:
+
+            if len(each_term.qubit_indices) == 2:
+                gate = gate_class(qubit_indices[0], qubit_indices[1])
+                gate.gate_label.update_gatelabel(
+                    new_application_sequence=two_qubit_count,
+                    new_gatemap_type=gatemap_type,
+                )
+                output_gates.append(gate)
                 two_qubit_count += 1
-
-            elif each_term.pauli_str in ['X', 'Y', 'Z']:
-
-                if each_term.pauli_str == 'X':
-                    output_gates.append(RXGateMap(each_term.qubit_indices[0],
-                                                    [*input_label, one_qubit_count]))
-                elif each_term.pauli_str == 'Y':
-                    output_gates.append(RYGateMap(each_term.qubit_indices[0],
-                                                    [*input_label, one_qubit_count]))
-                elif each_term.pauli_str == 'Z':
-                    output_gates.append(RZGateMap(each_term.qubit_indices[0],
-                                                    [*input_label, one_qubit_count]))
+            elif len(each_term.qubit_indices) == 1:
+                gate = gate_class(qubit_indices[0])
+                gate.gate_label.update_gatelabel(
+                    new_application_sequence=one_qubit_count,
+                    new_gatemap_type=gatemap_type,
+                )
+                output_gates.append(gate)
                 one_qubit_count += 1
 
         return output_gates
-    
-    def remap_gate_map_labels(gatemap_list: List[RotationGateMap], input_label: List) -> List[RotationGateMap]:
-        
+
+    def gatemaps_layer_relabel(
+        gatemap_list: List[GateMap], new_layer_number: int
+    ) -> List[RotationGateMap]:
         """
-        Recreates a list of RotationGateMap Objects with the appropriately 
-        assigned pauli_label attribute.
+        Reconstruct a new gatemap list from a list of RotationGateMap Objects with the input
+        layer number in the gate_label attribute.
+
+        Parameters
+        ----------
+        gatemap_list: `List[RotationGateMap]
+            The list of GateMap objects whose labels need to be udpated
         """
-        
-        output_gates = []
+        output_gate_list = []
 
-        one_qubit_count = 0
-        two_qubit_count = 0
-        
         for each_gatemap in gatemap_list:
-            
-            if each_gatemap.pauli_label[0] == '1q':
-                each_gatemap.pauli_label = [*input_label, one_qubit_count]
-                one_qubit_count += 1
-                
-            elif each_gatemap.pauli_label[0] == '2q':
-                each_gatemap.pauli_label = [*input_label, two_qubit_count]
-                two_qubit_count += 1
-                
-            output_gates.append(each_gatemap)
+            new_gatemap = deepcopy(each_gatemap)
+            new_gatemap.gate_label.update_gatelabel(new_layer_number=new_layer_number)
```

Comment:
Are we missing the assignment of `one_qubit_count` and `two_qubit_count` here?

Reply:
This function just updates the layer number attribute of the `gatelabel`. Without changing the one or two qubit counts. This is needed when we create layers in the circuit

Reply:
the assignment is now done within `QAOADescriptor` as part of the Custom Mixer fix.